### PR TITLE
Set ObjectFile's assignment operator to also be deleted like its copy constructor

### PR DESCRIPTION
**Pull request by fsfod**
This will also fix a warning when ObjectFile is dllexport'ed on Windows for my GSOC project for adding support Clang plugins on windows with @vgvassilev and @compnerd.

### CODE REVIEW

File: llvm/include/llvm/Object/ObjectFile.h [hunk 1]
```diff
@@ -302,6 +302,7 @@ class ObjectFile : public SymbolicFile {
 public:
   ObjectFile() = delete;
   ObjectFile(const ObjectFile &other) = delete;
+  ObjectFile operator=(const ObjectFile &other) = delete;
```
Comment (MaskRay) [inline llvm/include/llvm/Object/ObjectFile.h:305]: `ObjectFile &opeartor=`

**Review by MaskRay — APPROVED** [review]

**Review by JDevlieghere — APPROVED** [review]

**Comment by vgvassilev** [comment]
Can somebody land that for @fsfod. I do not think he has commit access.

**Comment by JDevlieghere** [comment]
I can merge it once the formatting issue has been addressed.

File: llvm/include/llvm/Object/ObjectFile.h (hunk 1 above)
Comment (dyung) [reply llvm/include/llvm/Object/ObjectFile.h]: This should probably be `&operator` and not `&opeartor`. I think this typo is causing nearly every buildbot to fail.

**Pull request merged**